dcos_installer/upgrades.py

```diff
@@ -0,0 +1,113 @@
+'''
+Generating node upgrade script
+'''
+import subprocess
+
+import gen.build_deploy.util as util
+import gen.template
+from dcos_installer.constants import SERVE_DIR
+from pkgpanda.build import hash_str
+from pkgpanda.util import write_string
+
+
+node_upgrade_template = """
+#!/bin/bash
+#
+# BASH script to upgrade DC/OS on a node
+#
+# Metadata:
+#   dcos image commit: {{ dcos_image_commit }}
+#   generation date: {{ generation_date }}
+# Copyright 2016 Mesosphere, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit -o nounset -o pipefail
+
+source /opt/mesosphere/environment.export
+
+# Check if this is a terminal, and if colors are supported, set some basic
+# colors for outputs
+if [ -t 1 ]; then
+    colors_supported=$(tput colors)
+    if [[ $colors_supported -ge 8 ]]; then
+    RED='\e[1;31m'
+    BOLD='\e[1m'
+    NORMAL='\e[0m'
+    fi
+fi
+
+if [[ $EUID -ne 0 ]]; then
+    echo "This script must be run as root" 1>&2
+    exit 1
+fi
+
+
+function main() {
+
+    # check for version of dc/os upgrading from
+    version=`grep "version" /opt/mesosphere/etc/dcos-version.json | cut -d '"' -f 4`
+    if [ $version != {{ dcos_version }} ]; then
+       echo "Not the correct version"
```

Comment:
This would be a pretty confusing error message to run into. This could be more verbose, like: `Expecting to upgrade from DC/OS version {{ dcos_version }}. Found $version.`